libcanard/canard_dsdl.c

```diff
@@ -31,8 +31,18 @@
 #    define CANARD_PRIVATE static
 #endif
 
-#if !defined(__STDC_VERSION__) || (__STDC_VERSION__ < 201112L)
-#    error "Unsupported language: ISO C11 or a newer version is required."
+#if !defined(__STDC_VERSION__) || (__STDC_VERSION__ < 199901L)
+#    error "Unsupported language: ISO C99 or a newer version is required."
+#endif
+
+/// In general, _Static_assert is not present on C99 compilers, except for gnu99
+#if (__STDC_VERSION__ < 201112L) && !defined(_Static_assert)
+// Intentional violation of MISRA: static assertion macro cannot be replaced with a function definition.
+#    define _Static_assert(x, ...) \
+        typedef char _static_assert_gl(_static_assertion_, __LINE__)[(x) ? 1 : -1]  // NOSONAR
+#    define _static_assert_gl(a, b) _static_assert_gl_impl(a, b)                    // NOSONAR
+// Intentional violation of MISRA: the paste operator ## cannot be avoided in this context.
+#    define _static_assert_gl_impl(a, b) a##b  // NOSONAR
```

Comment:
The problem here is that `_Static_assert` is a reserved name that cannot be redefined, no matter which version of the standard you're using (because any identifier that starts with an underscore followed by a capital letter is a reserved identifier). I thought Clang-Tidy was supposed to catch this and complain.

Maybe it's better to check `#ifndef static_assert` (sic! `static` not `_Static`) and then redefine that instead? Also replacing `_Static_assert` with `static_assert` in the sources.

The reason I'm using `_Static_assert` here is that the standard library that ships with AVR GCC does not define the standard macro `static_assert` for some reason (it's a bug). Since you will be explicitly checking for that, it would no longer be a problem.

Reply:
That's interesting. Given that kind of possible discrepancy for target-specific GCC compilers, wouldn't it be nice too to add some quick tests on other targets too? (ARM, xtensa, etc).

Reply:
Maybe, maybe not. The reason I singled out AVR is that it is an 8-bit architecture. If we could run tests against ARM or other platforms that would be highly valuable; testing whether it compiles at all is probably not very useful. In the end, if your compiler is not equipped with a standards-compliant library, it's your problem.